node-gui-x/src-tauri/src/lib.rs

```diff
@@ -0,0 +1,1045 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use chainstate::ChainInfo;
+use common::address::Address;
+use common::chain::{ DelegationId, PoolId };
+use common::primitives::Amount;
+use common::time_getter::TimeGetter;
+use node_gui_backend::messages::{
+    BackendEvent,
+    BackendRequest,
+    CreateDelegationRequest,
+    DecommissionPoolRequest,
+    DelegateStakingRequest,
+    EncryptionAction,
+    SendDelegateToAddressRequest,
+    SendRequest,
+    StakeRequest,
+    TransactionInfo,
+    WalletId,
+};
+use node_gui_backend::AccountId;
+use node_gui_backend::{ BackendSender, ImportOrCreate, InitNetwork, InitializedNode, WalletMode };
+use once_cell::sync::OnceCell;
+use serde::{ Deserialize, Serialize };
+use serde_json::Value;
+use wallet::account::transaction_list::TransactionList;
+use wallet_rpc_lib::types::{ Balances, PoolInfo };
+use std::collections::BTreeMap;
+use std::fmt::Debug;
+use std::path::PathBuf;
+use std::sync::Arc;
+use tauri::async_runtime::RwLock;
+use tauri::{ AppHandle, Emitter };
+use tokio::sync::mpsc::UnboundedReceiver;
+use wallet_types::wallet_type::WalletType;
+struct AppState {
+    initialized_node: RwLock<Option<InitializedNode>>,
+    backend_sender: RwLock<Option<BackendSender>>,
+    backend_receiver: RwLock<Option<UnboundedReceiver<BackendEvent>>>,
+    low_priority_backend_receiver: RwLock<Option<UnboundedReceiver<BackendEvent>>>,
+}
+
+static GLOBAL_APP_HANDLE: OnceCell<AppHandle> = OnceCell::new();
+
+#[derive(Debug, Serialize, Deserialize)]
+struct OpenCreateWalletRequest {
```

Comment:
please move all structs to a separate file (e.g. requests.rs). Ideally lib file should only contain modules decalration or `run` function in your case. Everything else should be decomposed into separate files